addons/mail/static/src/messaging/entity/composer/composer.js

```diff
@@ -172,20 +228,132 @@ function ComposerFactory({ Entity }) {
+        setFirstSuggestedPartnerActive() {
+            if (this._allSuggestedPartners[0]) {
+                this.update({ activeSuggestedPartner: [['link', this._allSuggestedPartners[0]]] });
+            }
+        }
+
+        setLastSuggestedPartnerActive() {
+            if (this._allSuggestedPartners.length > 0) {
+                this.update({
+                    activeSuggestedPartner: [['link', this._allSuggestedPartners[this._allSuggestedPartners.length - 1]]]
+                });
+            }
+        }
+
+        setNextSuggestedPartnerActive() {
+            if (this._allSuggestedPartners.length > 0) {
+                const activeSuggestedPartnerIndex = this._allSuggestedPartners.findIndex(
+                    suggestedPartner => suggestedPartner === this.activeSuggestedPartner
+                );
+                if (activeSuggestedPartnerIndex !== this._allSuggestedPartners.length - 1) {
+                    this.update({
+                        activeSuggestedPartner: [['link', this._allSuggestedPartners[activeSuggestedPartnerIndex + 1]]]
+                    });
+                } else {
+                    this.update({ activeSuggestedPartner: [['link', this._allSuggestedPartners[0]]] });
+                }
+            }
+        }
+
+        setPreviousSuggestedPartnerActive() {
+            if (this._allSuggestedPartners.length > 0) {
+                const activeSuggestedPartnerIndex = this._allSuggestedPartners.findIndex(
+                    suggestedPartner => suggestedPartner === this.activeSuggestedPartner
+                );
+                if (activeSuggestedPartnerIndex === -1) {
+                    this.update({
+                        activeSuggestedPartner: [['link', this._allSuggestedPartners[0]]]
+                    });
+                } else if (activeSuggestedPartnerIndex !== 0) {
+                    this.update({
+                        activeSuggestedPartner: [['link', this._allSuggestedPartners[activeSuggestedPartnerIndex - 1]]]
+                    });
+                } else {
+                    this.update({
+                        activeSuggestedPartner: [['link', this._allSuggestedPartners[this._allSuggestedPartners.length - 1]]]
+                    });
+                }
+            }
         }
 
         //----------------------------------------------------------------------
         // Private
         //----------------------------------------------------------------------
 
+        /**
+         *
+         * @private
+         */
+        _computeAllSuggestedPartners() {
+            return [['replace', this.mainSuggestedPartners.concat(this.extraSuggestedPartners)]];
+        }
+
+        /**
+         *
+         * @private
+         */
+        _computeCanPostMessage() {
+            return this.textInputContent || this.attachments.length !== 0;
+        }
+
+        /**
+         * Ensures to avoid duplicated partners in _allSuggestedPartners
```

Comment:
Recheck my previous comment, this doesn't ensure there is no duplicate in _allSuggestedPartners (that is taking care automatically because it is a relation), but instead that ensures there is no inconsistency in the rest of the code due to _allSuggestedPartners having no duplicates in the first place.